### DISCUSSION TITLE
Getting currentTime from ui.video

### DISCUSSION THREAD

**Discussion by dr-sandipan-chowdhury** [zauberzeug/nicegui#3278] [Ideas / Feature Requests]
How to get currentTime of the video playing in ui.video

**Comment by dr-sandipan-chowdhury** [comment]
Hi @rodja I would like to implement some functionalities to ui.video:
1. `get_current_time()` will return the current time in seconds
2. `on_pause(handler)` 
3. `on_play(handler)`
4. `on_seeked(handler)`

What do you think

**Reply by rodja** [reply]
Sounds good. Maybe instead of `get_current_time` you can provide a property `current_time`?

**Reply by dr-sandipan-chowdhury** [reply]
Yes. In that case we have to use a getter for the property so that `currentTime` could be accessed from the browser only when requested. This will prevent continuous `current_time` update whenever the video plays.

Also, a setter for `current_time` will look a more browserish version of `seek`.

What do you think?

**Reply by dr-sandipan-chowdhury** [reply]
Also, for `on_play` and other two methods, how about if the handler accept the current_time parameter? In that case, browser onplay will send the currentTime single handedly. 

I don't know how nicegui works under the hood... I downloaded the codebase and started playing around... Can you say where to start reading it so I can understand easily how nicegui works

**Reply by python-and-novella** [reply]
Maybe flet's methods can inspire you a little.
https://flet.dev/docs/controls/video/#methods
![image](https://github.com/zauberzeug/nicegui/assets/16256910/80ef7f37-4adf-41c6-8b95-b93228de30af)

**Comment by rodja** [comment]
Making current_time awaitable to only request it when needed is good. Also a setter property will come in handy. Passing `current_time` as parameter to the `on_`* methods is also good. And of course helpful methods/properties like `duration`, `is_playing` etc. are welcome too.

**Reply by BHM-Bob** [reply]
How is the progress now? I also need the `current_time` attribute from the `video`.

**Reply by falkoschindler** [reply]
@BHM-Bob The feature hasn't been implemented yet. And the feature request hasn't received any upvotes either.

But here is a workaround using `ui.run_javascript`:
```py
@ui.page('/')
def index():
    video = ui.video('https://test-videos.co.uk/vids/bigbuckbunny/mp4/h264/360/Big_Buck_Bunny_360_10s_1MB.mp4')

    async def get_current_time():
        time = await ui.run_javascript(f'getHtmlElement({video.id}).currentTime')
        ui.notify(f'Current time: {time:.2f} seconds')

    ui.button('Get current time', on_click=get_current_time)
```

**Reply by BHM-Bob** [reply]
Thank you for your helpful reply!
